controller/SynchronisationApi.php

```diff
@@ -81,22 +81,27 @@ public function fetchEntityChecksums()
     public function fetchEntityDetails()
     {
         try {
-            if ($this->getRequestMethod() != \Request::HTTP_GET) {
-                throw new \BadMethodCallException('Only GET method is accepted to access ' . __FUNCTION__);
+            if ($this->getRequestMethod() != \Request::HTTP_POST) {
+                throw new \BadMethodCallException('Only POST method is accepted to access ' . __FUNCTION__);
             }
 
-            if (!$this->hasRequestParameter(self::PARAM_TYPE)) {
+            $parameters = file_get_contents('php://input');
```

Comment:
```php
            $parameters = file_get_contents('php://input');
            if (!is_array($parameters = json_decode($parameters, true)) || json_last_error() !== JSON_ERROR_NONE) {
                throw new \InvalidArgumentException('Missing parameters to access ' . __FUNCTION__);
            }
```

This part of code it's duplicated in the method `fetchClassDetails` along with next if conditions, please move it to method to have on single point of truth.